lib/stats/tests/test_stats_cluster.c

```diff
@@ -163,6 +163,61 @@ Test(stats_cluster, test_stats_cluster_key_equal_when_custom_tags_are_the_same)
   stats_cluster_free(sc2);
 }
 
+Test(stats_cluster, test_stats_cluster_key_equal_with_labels)
```

Comment:
I don't see much difference between this test case and `test_stats_cluster_key`, other than creating `StatsCluster`s here.

I think they could be merged, and we could create `StatsCluster`s in the merged TC.

Also I think we should check the keys stored in the `StatsCluster`s, if we are creating them.